drivers/interrupt_controller/intc_plic.c

```diff
@@ -68,16 +68,35 @@
 #define INTC_PLIC_STATIC static inline
 #endif
 
+#ifdef CONFIG_PLIC_IRQ_AFFINITY
+#if CONFIG_MP_MAX_NUM_CPUS <= 8
+typedef uint8_t plic_cpumask_t;
+#elif CONFIG_MP_MAX_NUM_CPUS <= 16
+typedef uint16_t plic_cpumask_t;
+#elif CONFIG_MP_MAX_NUM_CPUS <= 32
+typedef uint32_t plic_cpumask_t;
+#elif CONFIG_MP_MAX_NUM_CPUS <= 64
+typedef uint64_t plic_cpumask_t;
```

Comment:
The API only specifies a uint32_t argument for the mask, so seems needless to try to support 64 bit here.  Or else we should widen the API.